action.yml

```diff
@@ -139,5 +140,47 @@ runs:
             DO_RELEASE=0
           fi
         fi
+
+        # Check again, this time using the double-dot syntax which will show the raw diff between the latest tag
+        # and the next-patch branch
+        # This isn't available via the github api, so screen scrape this instead. Screen scraping isn't
+        # great because it's brittle, however if this fails then all that happens is we tag a release that
+        # has no actual changes, which isn't the end of the world.
+        # Here we are only detecting if there are no actual changes to release, which can happen in a couple of scenarios:
+        # a) A change is made and tagged, and then backported to an older branch and then merged-up
+        # b) A change made in a previous major that we don't want to keep in current major, so
+        #    it's reverted during the merge-up
+        if [[ $DO_RELEASE == "1" ]]; then
+          RESP_CODE=$(curl -w %{http_code} -s -o __compare.html \
+          -X GET https://github.com/$GITHUB_REPOSITORY/compare/$LATEST_TAG..$GITHUB_SHA
+          )
+          if [[ $RESP_CODE != "200" ]]; then
+            echo "Unable to fetch compare html - HTTP response code was $RESP_CODE"
+            exit 1
+          fi
+          # Look for the following strings in the html
+          # `|| true` needs to be suffixed otherwise an error code of 1 will be omitted when there is no match
+          STRING_A_COUNT=$(cat __compare.html | grep -c "<strong>$LATEST_TAG</strong> and") || true
+          STRING_B_COUNT=$(cat __compare.html | grep -c "<strong>$GITHUB_SHA</strong> are identical.") || true
```

Comment:
Why are we counting these separately, instead of explicitly checking for the entire `"<strong>$LATEST_TAG</strong> and <strong>$GITHUB_SHA</strong> are identical."`?

Reply:
Because there's a line break in between them with a load of leading spaces for the second string

Reply:
I think this is too flimsy. They could add or remove whitespace at any time without affecting the way it renders in the browser. Might be better to do a quick `php -a` with `strip_tags` and do a whitespace-agnostic regex.